src/ops/fused.cpp

```diff
@@ -121,108 +127,112 @@ __host__ void FusedOp::forward_task(Task const *task,
-    Domain my_id[MAX_NUM_INPUTS];
-    Domain my_wd[MAX_NUM_WEIGHTS];
-    Domain my_od[MAX_NUM_OUTPUTS];
-    float const *my_ip[MAX_NUM_INPUTS];
-    float const *my_wp[MAX_NUM_WEIGHTS];
-    float *my_op[MAX_NUM_OUTPUTS];
+    GenericTensorAccessorR my_input_accessor[MAX_NUM_INPUTS];
+    GenericTensorAccessorR my_weight_accessor[MAX_NUM_WEIGHTS];
+    GenericTensorAccessorW my_output_accessor[MAX_NUM_OUTPUTS];
     for (int i = 0; i < fused->op_num_inputs[op]; i++) {
       int my_off = fused->op_input_idx[i + ioff];
       if (fused->op_input_source[i + ioff] == SOURCE_INPUT) {
-        my_id[i] = input_domain[my_off];
-        my_ip[i] = input_ptr[my_off];
+        my_input_accessor[i] = input_accessor[my_off];
       } else if (fused->op_input_source[i + ioff] == SOURCE_OUTPUT) {
-        my_id[i] = output_domain[my_off];
-        my_ip[i] = output_ptr[my_off];
+        my_input_accessor[i] = output_accessor[my_off];
       } else
         assert(false);
     }
     for (int i = 0; i < fused->op_num_weights[op]; i++) {
       assert(fused->op_weight_source[i + woff] == SOURCE_WEIGHT);
-      my_wd[i] = weight_domain[fused->op_weight_idx[i + woff]];
-      my_wp[i] = weight_ptr[fused->op_weight_idx[i + woff]];
+      my_weight_accessor[i] = weight_accessor[fused->op_weight_idx[i + woff]];
     }
     for (int i = 0; i < fused->op_num_outputs[op]; i++) {
       assert(fused->op_output_source[i + ooff] == SOURCE_OUTPUT);
-      my_od[i] = output_domain[fused->op_output_idx[i + ooff]];
-      my_op[i] = output_ptr[fused->op_output_idx[i + ooff]];
+      my_output_accessor[i] = output_accessor[i + ooff];
     }
     switch (fused->op_op_type[op]) {
       case OP_CONCAT: {
         assert(fused->op_num_weights[op] == 0);
         assert(fused->op_num_outputs[op] == 1);
         ConcatMeta *m = (ConcatMeta *)metas->meta[op];
         int num_inputs = fused->op_num_inputs[op];
-        Concat::forward_kernel(my_op[0],
-                               my_ip,
-                               num_inputs,
-                               m->legion_axis,
-                               my_od[0],
-                               my_id,
-                               stream);
+        Concat::forward_kernel_wrapper(m,
+                                       my_output_accessor[0],
+                                       my_input_accessor,
+                                       num_inputs,
+                                       m->legion_axis);
         break;
       }
       case OP_CONV2D: {
         assert(fused->op_num_inputs[op] == 1);
         assert(fused->op_num_outputs[op] == 1);
-        assert(my_id[0].get_dim() == 4);
-        assert(my_wd[0].get_dim() == 4);
-        assert(my_od[0].get_dim() == 4);
+        assert(my_input_accessor[0].domain.get_dim() == 5);
+        assert(my_weight_accessor[0].domain.get_dim() == 5);
+        assert(my_output_accessor[0].domain.get_dim() == 5);
         Conv2DMeta *m = (Conv2DMeta *)metas->meta[op];
```

Comment:
@lockshaw I copy pasted from fused.cu. I'm not familiar with what's being asserted and if it should still be from fused.cpp

Reply:
Yeah this is correct. This is a change from unity: we currently model replication with an additional replica dimension, so every tensor dimension internally ends up one bigger